sources/engine/Stride.Assets/StridePackageUpgrader.cs

```diff
@@ -228,13 +228,13 @@ public override bool UpgradeBeforeAssembliesLoaded(PackageLoadParameters loadPar
                             {
```

Comment:
Imho PackageUpdater should update from net5, net6, and net7 may be or no?

Reply:
for the net4 reference here is due to backwards compatability for .NET framework. This may not be needed anymore since this has changed since Stride 4.1. @Eideren would you be against removing this?

Reply:
Looks like we should be able to replace the whole
`if (dependency.Version.MinVersion < new PackageVersion("4.1.0.0") && solutionProject != null){...}`
with something like
`dependency.Version.MinVersion < this package version, swap to this tfm`
The additional checks for handwritten version seem unnecessary.

Reply:
What you'll want to do is bump to 4.2.0.1 in `sources\shared\SharedAssemblyInfo.cs`.
4.1 => net6.0
4.2 => net8.0

Then ideally, to ease maintenance, we want to keep the version upgrade atomic. This makes sure all upgrade paths are tested (since they're all run for older projects) and sequentially applied. Also, it's later easier to remove support for older version upgrades if later we say we want to support only upgrading from 4.1+ and 4.0 can't be opened anymore.

So what you can do is add a separate check after existing one, only for 4.1 to 4.2 (old code stay the same, then you add another `if (dependency.Version.MinVersion < new PackageVersion("4.2.0.0")` after the `if (4.1)` branch to do the upgrade to net8.0).

Reply:
I finally got around to updating this. I left out net7 since it was never officially supported anyway.